pkg/cidr: Move linux specific variable references from netlink

aditighag opened the pull request:
Commit e731cbe27e added references to linux specific consts from the netlink package, and it broke the ginkgo compilation on darwin targets - https://github.com/cilium/cilium/actions/runs/5944152105/job/16120804454?pr=27520. 

/cc @borkmann 

Reported-by: Tim Horner <timothy.horner@isovalent.com>
Fixes: e731cbe27e "cilium, iptables: Support option to masquerade source IP from routing layer"

aditighag committed:
pkg/cidr: Move linux specific variable references from netlink

The netlink consts are linux specific, so move them accordingly.

Reported-by: Tim Horner <timothy.horner@isovalent.com>
Fixes: e731cbe27e "cilium, iptables: Support option to masquerade source IP from routing layer"
Signed-off-by: Aditi Ghag <aditi@cilium.io>

joestringer commented on .github/workflows/lint-build-commits.yaml:
```diff
@@ -100,7 +100,7 @@ jobs:
       # Runs only if code under test/ is changed.
       - name: Check if ginkgo test suite build works for every commit
-        if: steps.test-tree.outputs.src == 'true'
+#        if: steps.test-tree.outputs.src == 'true'
```
Need to drop this before merging.

joestringer reviewed (approved):
Changes LGTM. I wonder if we also need something like this since we clearly have test code depending on `pkg/`:

```
diff --git a/.github/workflows/lint-build-commits.yaml b/.github/workflows/lint-build-commits.yaml
index 32d9fe21254d..c27c724f1ed3 100644
--- a/.github/workflows/lint-build-commits.yaml
+++ b/.github/workflows/lint-build-commits.yaml
@@ -96,6 +96,7 @@ jobs:
         with:
           filters: |
             src:
+              - 'pkg/**'
               - 'test/**'
 
       # Runs only if code under test/ is changed.
```

joestringer commented on pkg/cidr/cidr_unspecified.go:
```diff
@@ -0,0 +1,12 @@
+
+// Dummy values on non-linux platform
+const (
+	FAMILY_V4 = 0
```
```suggestion
	FAMILY_V4 = iota
```

thorn3r reviewed (approved):
thanks for the quick fix!

aditighag committed:
gha: Extend source directory for build runs

Previously, the GHA ran only for PRs with changes in the test directory.
However, this failed to catch build regressions in certain cases where PR changes
affected ginkgo builds.

Suggested-by: Joe Stringer <joe@cilium.io>
Signed-off-by: Aditi Ghag <aditi@cilium.io>

borkmann reviewed (commented):
Awesome, thanks for the fix!
